src/timeMachine/components/fluxFunctionsToolbar/FluxFunctionsToolbar.tsx

```diff
@@ -56,36 +55,22 @@ class FluxFunctionsToolbar extends PureComponent<Props, State> {
                     key={category}
                     category={category}
                     funcs={funcs}
-                    onClickFunction={this.handleClickFunction}
+                    onClickFunction={handleClickFunction}
                   />
                 ))
               }
             </TransformToolbarFunctions>
           </div>
         </DapperScrollbars>
-      </>
+      </ErrorBoundary>
     )
-  }
-
-  private handleSearch = (searchTerm: string): void => {
-    this.setState({searchTerm})
-  }
-
-  private handleClickFunction = (func: FluxToolbarFunction) => {
-    this.props.onInsertFluxFunction(func)
-  }
-}
-
-const mstp = (state: AppState) => {
-  const activeQueryText = getActiveQuery(state).text
-
-  return {activeQueryText}
+  }, [searchTerm, handleClickFunction])
 }
 
 const mdtp = {
   onSetActiveQueryText: setActiveQueryText,
 }
 
-const connector = connect(mstp, mdtp)
+const connector = connect(null, mdtp)
 
-export default connector(ErrorHandling(FluxFunctionsToolbar))
+export default connector(FluxFunctionsToolbar)
```

Comment:
nit: this was refactored to a functional component, why not use hooks here for dispatching the query?

Reply:
turns out there actually is no need for this function to be connected to redux at all so I just removed it